pkg/controller/controller.go

```diff
@@ -395,6 +398,12 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 		}
 	}
 
+	if p.enableNodeCheck && options.SelectedNode.Name != os.Getenv("NODE_NAME") {
```

Comment:
Isn't a node only selected for a PVC with late binding? What about PVCs without late binding? Are those not supported anymore when deploying the hostpath driver with provisioner on each node with node check enabled?

Reply:
Correct that will be a prereq to using this mode. Maybe we need to detect if they're not using delayed binding and return an error msg

Reply:
We could enable leadership election in this mode and then provision volumes which don't have a node chosen on the node which has the leader. I suspect that the csi-driver-host-path will not pass driver testing in this mode unless it still supports volumes with immediate binding.

If we combine that with yielding leadership after each provisioned volume, then such volumes would even be spread across the entire cluster. Not sure how important that is.

Reply:
is there a TLDR or doc on "immediate" and "late" binding in this context somewhere? I don't think I've seen these referenced in the context of kubernetes.

Reply:
https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode

Sorry I wasn't using the official terminology. "late" = "WaitForFirstConsumer"

I think using leader election to support the "immediate" mode is an interesting idea, but it's going to be really hard to support both immediate and waitforfirstconsumer at the same time, because we don't want leader election in the latter case.

For now, I am fine with saying that you must use waitforfirstconusmer binding mode if you run the hostpath driver like this.

Reply:
> For now, I am fine with saying that you must use waitforfirstconusmer binding mode if you run the hostpath driver like this.

Yes, might be good enough. Then a better error message when it encounters a volume without node name would be nice, though. Right now the resulting error message will be `Selected node () is not current node (xxxx)` (empty first bracket), which will be a bit misleading.

Anyway, to get a better understanding of how complicated it would be, I implemented my idea. The result is here: https://github.com/pohly/external-provisioner/commit/df7b61e39018cf3f8e5736b667965d8ead78414c

I've not tested whether it actually works and I don't intend to pursue this further unless there is interest in this approach. The biggest concern that I have myself is about how costly leadership election is when many processes are involved. No idea how "many" will be "too many"...

/cc @andrewsykim